sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVInferSchema.scala

```diff
@@ -233,7 +237,39 @@ class CSVInferSchema(val options: CSVOptions) extends Serializable {
    * is compatible with both input data types.
    */
   private def compatibleType(t1: DataType, t2: DataType): Option[DataType] = {
-    TypeCoercion.findTightestCommonType(t1, t2).orElse(findCompatibleTypeForCSV(t1, t2))
+    (t1, t2) match {
```

Comment:
Should this match be in `findCompatibleTypeForCSV`? Or does `findTightestCommonType` merge DateType and TimestampType in a way that is not applicable here?

Reply:
`findTightestCommonType` merge DateType and TimestampType in a way that is not applicable here?

Reply:
What result does `findTightestCommonType` return for DateType and TimestampType?

Reply:
```
(d1, d2) match {
      case (_: TimestampType, _: DateType) | (_: DateType, _: TimestampType) =>
        TimestampType

      case (_: TimestampType, _: TimestampNTZType) | (_: TimestampNTZType, _: TimestampType) =>
        TimestampType

      case (_: TimestampNTZType, _: DateType) | (_: DateType, _: TimestampNTZType) =>
        TimestampNTZType
    }
```

Reply:
Never mind, I checked the code, the resulting type will be TimestampType or TimestampNTZ.